### PR TITLE
Adds Truss for Mixtral 8x7B using vLLM

### PR DESCRIPTION
**Pull request by varunshenoy**

### CODE REVIEW

File: mistral/mixtral-8x7b-instruct-vllm/config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+base_image:
```
Comment (bolasim) [inline mistral/mixtral-8x7b-instruct-vllm/config.yaml:1]: curious why we need this base image?

File: mistral/mixtral-8x7b-instruct-vllm/config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  image: nvcr.io/nvidia/pytorch:23.11-py3
+  python_executable_path: /usr/bin/python3
+environment_variables: {}
+external_package_dirs: []
+model_metadata: {}
+model_name: Mixtral 8x7B vLLM
+python_version: py310
```
Comment (bolasim) [inline mistral/mixtral-8x7b-instruct-vllm/config.yaml:8]: drop this is using base image

File: mistral/mixtral-8x7b-instruct-vllm/config.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+base_image:
+  image: nvcr.io/nvidia/pytorch:23.11-py3
+  python_executable_path: /usr/bin/python3
+environment_variables: {}
```
Comment (bolasim) [inline mistral/mixtral-8x7b-instruct-vllm/config.yaml:4]: let's drop empty ones too if possible.

File: mistral/mixtral-8x7b-instruct-vllm/model/model.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+
+
+class Model:
+    def __init__(self, **kwargs):
+        self.model = None
+        self.llm_engine = None
+
+        # Install megablocks and run ray process
+        subprocess.run(["pip", "install", "megablocks"])
```
Comment (bolasim) [inline mistral/mixtral-8x7b-instruct-vllm/model/model.py:15]: why install this here instead of requirements?

**Review by bolasim — APPROVED** [review]

File: mistral/mixtral-8x7b-instruct-vllm/model/model.py (hunk 4 above)
Comment (varunshenoy) [reply mistral/mixtral-8x7b-instruct-vllm/model/model.py:15]: `megablocks` needs a GPU build machine. This is a temp workaround.

File: mistral/mixtral-8x7b-instruct-vllm/config.yaml (hunk 1 above)
Comment (varunshenoy) [reply mistral/mixtral-8x7b-instruct-vllm/config.yaml:1]: base image has `nvcc` and the right version of `torch` installed. There was a lot of messing around and this was the only configuration that didn't lead to a failed build of `megablocks`.

**Pull request merged**